tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_all_opcodes_in_container.py

```diff
@@ -129,3 +131,271 @@ def test_all_opcodes_in_container(
         data=eof_code,
         expect_exception=expect_exception,
     )
+
+
+@pytest.mark.parametrize(
+    "opcode",
+    [
+        op
+        for op in list(Op)
+        if op not in invalid_eof_opcodes
+        and op not in halting_opcodes
+        and op not in [Op.RJUMP, Op.RJUMPI, Op.RJUMPV]
+    ],
+)
+def test_all_invalid_terminating_opcodes(
+    eof_test: EOFTestFiller,
+    opcode: Opcode,
+):
+    """
+    Test all opcodes that are invalid as the last opcode in a container
+    """
+    if opcode.has_data_portion():
+        opcode_with_data_portion = opcode[0] if opcode != Op.CALLF else opcode[1]
+    else:
+        opcode_with_data_portion = opcode
+
+    bytecode = (Op.ORIGIN * opcode_with_data_portion.min_stack_height) + opcode_with_data_portion
+
+    sections = [Section.Code(code=bytecode)]
+
+    if opcode == Op.CALLF:
+        sections += [Section.Code(code=Op.RETF, code_outputs=0)]
+    elif opcode == Op.EOFCREATE:
+        sections += [
+            Section.Container(
+                container=Container(
+                    sections=[
+                        Section.Code(code=Op.RETURNCONTRACT[0](0, 0)),
+                        Section.Container(Container.Code(code=Op.STOP)),
+                    ]
+                )
+            )
+        ]
+
+    sections += [Section.Data(b"\0" * 32)]
+
+    eof_test(
+        data=Container(
+            sections=sections,
+        ),
+        expect_exception=EOFException.MISSING_STOP_OPCODE,
+    )
+
+
+@pytest.mark.parametrize(
+    "opcode",
+    [op for op in list(Op) if op in halting_opcodes],
+)
+def test_all_unreachable_terminating_opcodes_after_stop(
+    eof_test: EOFTestFiller,
+    opcode: Opcode,
+):
+    """
+    Test all opcodes terminating opcodes after stop.
+    """
+    match opcode:
+        case Op.STOP:
+            sections = [Section.Code(code=Op.STOP + Op.STOP)]
+        case Op.RETF:
+            sections = [
+                Section.Code(code=Op.STOP + Op.CALLF[1]),
```

Comment:
This will fail validation on the first code section.
```suggestion
                Section.Code(code=Op.CALLF[1]),
```